### PR TITLE
 New sumo - do not merge

### PR DESCRIPTION
**Pull request by eugenevinitsky**

### CODE REVIEW

File: flow/controllers/car_following_models.py [hunk 1]
```diff
@@ -381,11 +381,22 @@ def get_accel(self, env):
...
-        # junctions. Setting them to 0 causes vehicles to not move; therefore,
-        # we maintain these negative headways to let sumo control the dynamics
-        # as it sees fit at these points.
+        edge = env.vehicles.get_edge(self.veh_id)
+        lead_edge = env.vehicles.get_edge(lead_id)
+
+        # This covers situations where both vehicles are in a junction from
+        # either direction (e.g. in the figure eight). The vehicle that is
+        # deeper into the junction has its accelerations defined by sumo. This
+        # needed is due to modifications in sumo from 1d4338ab80 to b6f37dd
```
Comment (eugenevinitsky) [inline flow/controllers/car_following_models.py]: commit number in this line does not seem correct? should be 016c09d306

Comment (eugenevinitsky) [reply flow/controllers/car_following_models.py]: Also, why would you want only the lead vehicle to have its acceleration defined by sumo? Why not both?

Comment (AboudyKreidieh) [reply flow/controllers/car_following_models.py]: As in have the acceleration of vehicles in a junction be defined by sumo? I can try that

Comment (eugenevinitsky) [reply flow/controllers/car_following_models.py]: I mean, you're already doing it for the lead vehicle?

Comment (AboudyKreidieh) [reply flow/controllers/car_following_models.py]: I actually like your idea better, the solution is more cleaner and more elegant, and the behavior is the same. Will push changes now

**Comment by eugenevinitsky** [comment]
Current errors that must be resolved before merge
- [] bottleneck rllib experiments do not run
- [] the sumo grid example has conflicting directions for the traffic lights
- [] cars go black in the bay bridge as they pass the toll light

**Comment by eugenevinitsky** [comment]
Grid: traffic phases must now be GrGr and yryr instead of the 12 digit strings from before

**Comment by AboudyKreidieh** [comment]
there is a one step lag for adding/removing subscriptions and reading data from the subscriptions. Modifications to the vehicles class will resolve the reason why the bottleneck experiment was not running

**Comment by AboudyKreidieh** [comment]
Update: I modified the vehicles class to immediately return subscription results from new vehicles. This resolved the "index out of range" problem in the bay bridge.

**Comment by eugenevinitsky** [comment]
Bottleneck capacity curve:
image

**Comment by AboudyKreidieh** [comment]
merged relevant changes to `new_sumo_pr`, so closing

**Pull request closed without merging**